src/backend/crossterm.rs

```diff
@@ -4,28 +4,140 @@
+            CMouseButton::Left => MouseButton::Left,
+            CMouseButton::Right => MouseButton::Right,
+            CMouseButton::Middle => MouseButton::Middle,
+        }
+    }
+}
+
+impl From<KeyCode> for Key {
+    fn from(code: KeyCode) -> Self {
+        match code {
+            KeyCode::Esc => Key::Esc,
+            KeyCode::Backspace => Key::Backspace,
+            KeyCode::Left => Key::Left,
+            KeyCode::Right => Key::Right,
+            KeyCode::Up => Key::Up,
+            KeyCode::Down => Key::Down,
+            KeyCode::Home => Key::Home,
+            KeyCode::End => Key::End,
+            KeyCode::PageUp => Key::PageUp,
+            KeyCode::PageDown => Key::PageDown,
+            KeyCode::Delete => Key::Del,
+            KeyCode::Insert => Key::Ins,
+            KeyCode::Enter => Key::Enter,
+            KeyCode::Tab => Key::Tab,
+            KeyCode::F(n) => Key::from_f(n),
+            KeyCode::BackTab => Key::Tab, /* not supported */
+            KeyCode::Char(_) => Key::Tab, /* is handled at `Event` level, use tab as default */
+            KeyCode::Null => Key::Tab, /* is handled at `Event` level, use tab as default */
+        }
+    }
+}
+
+impl From<CKeyEvent> for Event {
+    fn from(event: CKeyEvent) -> Self {
+        const CTRL_ALT: KeyModifiers = KeyModifiers::from_bits_truncate(
+            KeyModifiers::CONTROL.bits() | KeyModifiers::ALT.bits(),
+        );
+        const CTRL_SHIFT: KeyModifiers = KeyModifiers::from_bits_truncate(
+            KeyModifiers::CONTROL.bits() | KeyModifiers::SHIFT.bits(),
+        );
+        const ALT_SHIFT: KeyModifiers = KeyModifiers::from_bits_truncate(
+            KeyModifiers::ALT.bits() | KeyModifiers::SHIFT.bits(),
+        );
+
+        match event {
+            CKeyEvent {
+                modifiers: KeyModifiers::CONTROL,
+                code: KeyCode::Char(c),
+            } => Event::CtrlChar(c),
+            CKeyEvent {
+                modifiers: KeyModifiers::ALT,
+                code: KeyCode::Char(c),
+            } => Event::AltChar(c),
+            CKeyEvent {
+                modifiers: _,
+                code: KeyCode::Char('c'),
+            } => Event::Exit,
+
+            CKeyEvent {
+                modifiers: CTRL_ALT,
+                code,
+            } => Event::CtrlAlt(Key::from(code)),
+            CKeyEvent {
+                modifiers: CTRL_SHIFT,
+                code,
+            } => Event::CtrlShift(Key::from(code)),
+            CKeyEvent {
+                modifiers: ALT_SHIFT,
+                code,
+            } => Event::AltShift(Key::from(code)),
+
+            CKeyEvent {
+                modifiers: KeyModifiers::CONTROL,
+                code,
+            } => Event::Ctrl(Key::from(code)),
+            CKeyEvent {
+                modifiers: KeyModifiers::ALT,
+                code,
+            } => Event::Alt(Key::from(code)),
+            CKeyEvent {
+                modifiers: KeyModifiers::SHIFT,
```

Comment:
When writing uppercase letters, will the modifier be `SHIFT`?

Reply:
No, when you enabled capslock, then no SHIFT is sent. Only an  uppercase character.

Reply:
What about pressing <kbd>Shift</kbd>+<kbd>A</kbd> to send a single uppercase `'A'`?

Reply:
That is correct, I was not sure how to handle this case. Because cursive doesn't seem to have the `ShiftChar()` field. How is this handled currently? I would like to support SHIFT + CHAR as well.

Reply:
In cursive it's just a `Event::Char` with the uppercase character, like `Event::Char('A')` vs `Event::Char('a')`.

Reply:
Ah, okay, I can add that behaviour